sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java

```diff
@@ -244,4 +286,255 @@ public RegistryImpl disablePublicNetworkAccess() {
         }
         return this;
     }
+
+    @Override
+    public RegistryImpl withAccessFromSelectedNetworks() {
+        ensureNetworkRuleSet();
+        if (isInCreateMode()) {
+            this.innerModel().networkRuleSet().withDefaultAction(DefaultAction.DENY);
+        } else {
+            updateParameters.networkRuleSet().withDefaultAction(DefaultAction.DENY);
+        }
+        return this;
+    }
+
+    @Override
+    public RegistryImpl withAccessFromAllNetworks() {
+        ensureNetworkRuleSet();
+        if (isInCreateMode()) {
+            this.innerModel().networkRuleSet().withDefaultAction(DefaultAction.ALLOW);
+        } else {
+            updateParameters.networkRuleSet().withDefaultAction(DefaultAction.ALLOW);
+        }
+        return this;
+    }
+
+    @Override
+    public RegistryImpl withAccessFromIpAddressRange(String ipAddressCidr) {
+        ensureNetworkRuleSet();
+        if (this.innerModel().networkRuleSet().ipRules()
+            .stream().noneMatch(ipRule -> Objects.equals(ipRule.ipAddressOrRange(), ipAddressCidr))) {
+            this.innerModel().networkRuleSet().ipRules().add(new IpRule().withAction(Action.ALLOW).withIpAddressOrRange(ipAddressCidr));
+        }
+        if (!isInCreateMode()) {
+            updateParameters.networkRuleSet().withIpRules(this.innerModel().networkRuleSet().ipRules());
```

Comment:
Does it need to also check for duplicate item? User can "with" same trice in update?

Reply:
Yes. Deduplicate and add item is done in previous step. When updating, we just copy networkRuleSet from `innerModel`'s.

Added a test step for verifying deduplication.